### PR TITLE
🦋 Implement empty state in inbox

### PR DESCRIPTION
**Pull request by josemigallas**
[THREESCALE-9856: Implement EmptyState (Inbox)](https://issues.redhat.com/browse/THREESCALE-9856)

Before:
Screenshot 2023-10-10 at 11 02 38

After:
Screenshot 2023-10-11 at 09 08 41

### CODE REVIEW

File: app/views/provider/admin/messages/inbox/index.html.slim [hunk 1]
```diff
@@ -1,24 +1,32 @@
 - content_for :page_header_title, 'Inbox'
 
-= render 'provider/admin/messages/bulk_operations', scope: :received_messages, messages: @messages
+- if @messages.blank?
+  div class="pf-c-empty-state"
+    div class="pf-c-empty-state__content"
+      i class="fas fa-rocket pf-c-empty-state__icon" aria-hidden="true"
```
Comment (mayorova) [inline app/views/provider/admin/messages/inbox/index.html.slim:6]: I'm suggesting :inbox_tray: 
https://fontawesome.com/v4/icon/inbox

Comment (josemigallas) [reply app/views/provider/admin/messages/inbox/index.html.slim:6]: Not saying I don't like it, but Patternfly seems to be rather strict about icons in empty states: 
>To maintain consistency, specific icons are mapped to specific empty state scenarios.

https://www.patternfly.org/components/empty-state/design-guidelines#icons

I will ask in their Slack

Comment (mayorova) [reply app/views/provider/admin/messages/inbox/index.html.slim:6]: yeah, but according to that page, :rocket: is for getting-started, which is not the case

also it says "You can replace this icon if you already have an application-specific graphic that you want to use instead." :grimacing: 

but if we stick to that info, looks like :mag: is the one to use.
But I'd still prefer the icons I suggested :sweat_smile:

Comment (josemigallas) [reply app/views/provider/admin/messages/inbox/index.html.slim:6]: I confirmed with them these are just suggestions so we go with the inbox icon.

File: config/locales/en.yml [hunk 2]
```diff
@@ -428,7 +428,9 @@ en:
             success: Messages moved into the trash
         no_metrics: "You have no metrics"
         no_methods: "You have no methods"
-        no_received_messages: "You have no messages."
+        empty_state:
+          title: "You have no messages."
+          body: "Your inbox is empty. Send a message to your users to get started."
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply config/locales/en.yml:433]: > Then you can just use `t('.empty_state.title')` and `t('.empty_state.body')` in the view.
> 

I think I tried this in Outbox and didn't work, but I will give it another try.

> Also (regarding the "body" text), sending a message will not actually make anything new appear in the inbox, but if it's a requirement to have a button in the "empty state", then I guess it's fine - also, given that there is a "compose" button on the Inbox messages table as well.

True. TBH I just copied the text from some empty state that I found hehe. It's not a requirement to have a primary action, although I kinda like it (it makes que empty state no so... empty) and it's also totally fine to have the same action twice from a Patternfly perspective, it seems:

Screenshot 2023-10-11 at 08 46 19

I'll see if I can think of a better body message.

**Review by mayorova — APPROVED** [review]

**Pull request merged**